Classes/Toolbar/FLEXExplorerToolbar.m

```diff
@@ -150,9 +138,31 @@ - (void)layoutSubviews
     descriptionLabelFrame.size.width = CGRectGetMaxX(self.selectedViewDescriptionContainer.bounds) - kHorizontalPadding - descriptionOriginX;
     self.selectedViewDescriptionLabel.frame = descriptionLabelFrame;
 }
+    
+    
+#pragma mark - Setter Overrides
 
+- (void)setToolbarItems:(NSArray *)toolbarItems {
+    // Remove old toolbar items, if any
+    for (FLEXToolbarItem *item in _toolbarItems) {
+        [item removeFromSuperview];
+    }
+    
+    // Trim to 5 items if necessary
+    if (toolbarItems.count > 5) {
+        toolbarItems = [toolbarItems subarrayWithRange:NSMakeRange(0, 5)];
+    }
 
-#pragma mark - Setter Overrides
+    for (FLEXToolbarItem *item in toolbarItems) {
+        [self addSubview:item];
+    }
+
+    _toolbarItems = toolbarItems.copy;
+
+    // Lay out new items
+    [self setNeedsLayout];
+    [self layoutIfNeeded];
```

Comment:
I think it makes sense for the UI to update as soon as you set the property. This is how all UIKit views behave (UITabBar.items, UINavigationItem.rightBarButtonItems, etc). There's probably never a time one wouldn't want the UI to update right away.

Imo the cost of updating the UI prematurely is negligible, but if you want I could just set the ivar directly in the initializer and add the subviews there too